### ISSUE TITLE
Implement PQ-MIM compression paper

### ISSUE THREAD

**Issue by mmuckley** [facebookresearch/NeuralCompression#189]
We would like to have an implementation of the following paper:

[Image Compression with Product Quantized Masked Image Modeling](https://arxiv.org/abs/2212.07372)
Alaaeldin El-Nouby, Matthew J. Muckley, Karen Ullrich, Ivan Laptev, Jakob Verbeek, and Hervé Jégou

**Comment by aditya-prayaga** [comment]
Hey @mmuckley is this still available to pick up. I am Aditya Prayaga, a MS in CS student from NEU, Boston. For my ML class by Professor Paul Hand, I either have to pick a project or implement a base line of research implementation. So, I wanted to pick this up and start working on this. This would a great plus if I implemented for my PHD portfolio.

**Comment by mmuckley** [comment]
Hello @aditya-prayaga, it's certainly still available. If you do it, please try to use practices from the other project implementations in the repository for your implementation.

**Comment by aditya-prayaga** [comment]
Hey @mmuckley,

This is my first time working individually on a research implementation. I would greatly appreciate any help I get implementing this.

Before I implement or start anything I wanted to List the tasks here for your confirmations. So that my understanding is correct and also validates my understanding of the research paper.

# Understanding of Research Paper
We are Compressing Image using product quantization and masked image modeling (MIM) techniques. The method involves dividing the image into small subvectors, quantizing each subvector independently, and using a conditional entropy model based on MIM to improve the compression performance. The resulting compressed image has a smaller size and improved quality when uncompressed. The proposed method is compared to several strong neural compression baselines and achieves competitive performance in terms of PSNR, MS-SSIM, and FID metrics.

I started with 2 tasks

1) Setting Up Repo
- Forked it
- Created a folder named "pq_mim_compression" in projects
- Created a venv and added requirements file with flake8, isort and black packages.
- Create a readme and seeing other projects about how the content description is done.

2) Steps in Preprocessing
- I will create a config folder and add data conf folder where the yaml config is stored.
- The question i have is, The paper mentions using ImageNet data for pre-training and training and a percent of CIFAR-10 dataset to training and testing evaluating right?

Any suggestions on next steps are greatly appreciated. Thanks for your time on helping me.
